import/index_java/src/org/vufind/index/GeoTools.java

```diff
@@ -280,9 +378,13 @@ protected boolean validateDDCoordinates(Record record, Double west, Double east,
     */
    public boolean validateLines(Record record, Double west, Double east, Double north, Double south) {
     if ((!west.equals(east) && north.equals(south)) && (north == 90 || south == -90)) {
-        ControlField recID = (ControlField) record.getVariableField("001");
-        String recNum = recID.getData();
-        logger.error("Record ID: " + recNum.trim() + " - Coordinates form a line at the pole");
+        String msgError = "Coordinates form a line at the pole";
+        HashMap<Character, String> coords = new HashMap();
```

Comment:
This hashmap creation is repeated a lot of times. Maybe you need a helper method that just takes west, east, north, south and builds the map.

Reply:
Great idea. I'll implement that. Thanks!